warehouse/query-core/src/test/java/datawave/query/jexl/visitors/BoundedRangeDetectionVisitorTest.java

```diff
@@ -0,0 +1,56 @@
+package datawave.query.jexl.visitors;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import datawave.data.type.StringType;
+import datawave.data.type.Type;
+import datawave.query.config.ShardQueryConfiguration;
+import datawave.query.exceptions.DatawaveFatalQueryException;
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.util.MockMetadataHelper;
+import org.apache.commons.jexl2.parser.ASTJexlScript;
+import org.apache.commons.jexl2.parser.ParseException;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class BoundedRangeDetectionVisitorTest {
+    private ShardQueryConfiguration config;
+    private MockMetadataHelper helper;
+    
+    @Before
+    public void init() throws Exception {
+        this.config = ShardQueryConfiguration.create();
+        Multimap<String,Type<?>> queryFieldsDatatypes = HashMultimap.create();
+        queryFieldsDatatypes.put("foo", new StringType());
+        
+        config.setQueryFieldsDatatypes(queryFieldsDatatypes);
+        
+        this.helper = new MockMetadataHelper();
+        this.helper.setNonEventFields(queryFieldsDatatypes.keySet());
+    }
+    
+    @Test
+    public void testBoundedRange() throws ParseException {
+        String queryString = "((_Bounded_ = true) && (foo >= '1' && foo <= '10'))";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(queryString);
+        assertTrue(BoundedRangeDetectionVisitor.mustExpandBoundedRange(config, helper, script));
```

Comment:
I'm always a fan of delegating to a test method so you can easily test multiple queries per method. 

At a minimum I would like to see two more tests added where the bounded range is the child of an OR and the child of an AND. For example `FOO == 'bar' && <bounded range>` and `FOO == 'bar' || <bounded range>`

Reply:
Added as requested.